proposals/NNNN-isolated-any-functions.md

```diff
@@ -0,0 +1,671 @@
+  initialized with a derivation of `E`; or
+- it is a non-optional derivation of `E`.
+
+When analyzing an expression for these rules, certain non-instrumental
+differences in expression syntax and behavior must be ignored.  This
+and certain other terms in this section are intended to match the
+definitions laid out in [SE-0420](https://github.com/apple/swift-evolution/blob/main/proposals/0420-inheritance-of-actor-isolation.md#generalized-isolation-checking).
+
+For example:
+
+```swift
+func delay(operation: @isolated(any) () -> ()) {
+  let isolation = operation.isolation
+  Task { [isolated isolation] in
+    print("waking")
+    operation() // <-- does not cross an isolation barrier and so is synchronous
+    print("finished")
+  }
+}
+```
+
+(This uses the `isolated` capture modifier proposed in the draft
+[closure isolation control][isolated-captures] proposal.  It would be
+possible but significantly more awkward to get this effect with that
+proposal.)
+
+### Adoption in task-creation routines
+
+There are a large number of functions in the standard library that create
+tasks:
+- `Task.init`
+- `Task.detached`
+- `TaskGroup.addTask`
+- `TaskGroup.addTaskUnlessCancelled`
+- `ThrowingTaskGroup.addTask`
+- `ThrowingTaskGroup.addTaskUnlessCancelled`
+- `DiscardingTaskGroup.addTask`
+- `DiscardingTaskGroup.addTaskUnlessCancelled`
+- `ThrowingDiscardingTaskGroup.addTask`
+- `ThrowingDiscardingTaskGroup.addUnlessCancelled`
+
+This proposal modifies all of these APIs so that the task function has
+`@isolated(any)` function type.  These APIs now all synchronously enqueue
+the new task directly on the appropriate executor for the task function's
+dynamic isolation.
+
+Swift reserves the right to optimize the execution of tasks to avoid
+"unnecessary" isolation changes, such as when an isolated `async` function
+starts by calling a function with different isolation.[^2] In general, this
+includes optimizing where the task initially starts executing:
+
+```swift
+@MainActor class MyViewController: UIViewController {
+  @IBAction func buttonTapped(_ sender : UIButton) {
+    Task {
+      // This closure is implicitly isolated to the main actor, but Swift
+      // is free to recognize that it doesn't actually need to start there.
+      let image = await downloadImage()
+      display.showImage(image)
+    }
+  }
+}
+```
+
+[^2]: This optimization doesn't change the formal isolation of the functions
+involved and so has no effect on the value of either `#isolation` or
+`.isolation`.
+
+As an exception, in order to provide a primitive scheduling operation with
+stronger guarantees, Swift will always start a task function on the
+appropriate executor for its formal dynamic isolation unless:
+- it is non-isolated or
+- it comes from a closure expression that is only *implicitly* isolated
+  to an actor (that is, it has neither an explicit `isolated` capture
+  nor a global actor attribute).  This can currently only happen with
+  `Task {}`.
```

Comment:
This came up in the [pitch thread](https://forums.swift.org/t/isolated-any-function-types/70562/8), and you [answered it there](https://forums.swift.org/t/isolated-any-function-types/70562/10). The proposal would be stronger if it included this justification.